dojo/utils.py

```diff
@@ -1834,3 +1834,60 @@ def create_bleached_link(url, title):
     link += title
     link += '</a>'
     return bleach.clean(link, tags=['a'], attributes={'a': ['href', 'target', 'title']})
+
+
+def auto_delete_engagements():
+    """
+    For an engagement to be in-scope for automated deletion, the following rules apply:
+    - must have been updated before x days (as defined in system settings)
+    - (hardcoded) must be a CI/CD engagement
+    - (hardcoded) must only contain duplicate findings
+    - (hardocded) must not contain any notes on any of its findings
+
+    The original use-case of this feature relates to the mass imports that one can have through CI pipelines,
+    generating a vast amount of findings which ultimately will boggle down defectdojo's performance
+    and make it harder to see what needs to be seen.
+    """
+    def _notify(engagement_id, engagement_title):
+        create_notification(
+            event='auto_delete_engagement',
+            title=engagement_title,
+            id=engagement_id,
+        )
+
+    system_settings = System_Settings.objects.get()
+    if system_settings.engagement_auto_delete_enable:
+        # how to not exclude the tag when not empty? If empty, then query results are unexpected.
+        # setting arbitrary string for now, which is unlikely to be a used tag.
+        lock_tag = system_settings.engagement_auto_delete_lock_tag or 'qAEH2HL6Qd9ofZYLCGykN2WQ'
```

Comment:
@valentijnscholten aside your regular feedback, a quick tip on that?

Reply:
what's the question